sale_coupon_advanced/models/sale_order.py

```diff
@@ -112,3 +112,47 @@ def _update_pricelist(self, pricelist):
         self.pricelist_id = pricelist
         for line in self.order_line:
             line.product_id_change()
+
+    def write(self, vals):
```

Comment:
Unfortunately its not that simple. First iteration implementation was done just like that. But there are side effects that happen when you try to unlink lines via Odoo interface (Form). It tries to use same line that is being deleted (by onchange spec). So we decided to make workaround that we clean up write if such things happens. You can check tests, specifically the one with Form. Thats special case, where all this complexity originates from.

Reply:
@StephaneMangin if by normal procedure you mean `recompute coupon lines`, then it wont work. As you already found my comment, it works the opposite of what we need. It adds missing lines, instead of removing. So in that case, we won't know if lines need to be added or removed.